File: frontend/src/game/messages.js
export default {
  title: 'TIC TAC TOE',
  intro:
    'Win untold glory by successfully placing a complete line in any horizontal, vertical or diagonal direction.',
  instructions:
    'Choose a number between 1 and 9 on alternate turns to place your mark in the 3x3 grid.',
  nan: "Sorry, that doesn't seem to be a valid number. Please try again: ",
  outOfBounds:
    "Sorry, that number isn't available on the board. Please try again: ",
  occupied: 'Dang - your opponent got there before you! Try again: ',
  draw: "How unsatisfying! It's a draw!",
  unknownError: "Well, something's gone wrong somewhere.",
  winner: mark => `Player ${mark} wins!`,
  turn: mark => `Player ${mark}'s turn`,
}

File: cli/src/main/java/Messages.java
public class Messages {
    static String setupInstructions() {
        return "Select a game mode:\n1. Human v Human\n2. Human v CPU (EASY)\n3. Human v CPU (HARD)\n4. CPU v CPU (EASY)\n5. CPU v CPU (HARD)";
    }

    static String boardSetupInstructions() {
        return "Select a board size:\n1. 3x3 (default)\n2. 4x4";
    }

    static String invalidMove() {
        return "Sorry, that move is invalid.\nPlease choose an unoccupied square.";
    }

    static String playerWin(Player player) {
        return String.format("Player %s wins!", player.symbol().toString());
    }

    static String playersDraw() {
        return "Bad luck! It's a draw!";
    }

    static String announcePlayerTurn(Player player) {
        return String.format("Player %s's turn", player.symbol().toString());
    }

    static String getInstructions(int sideLength) {
        String gridShape = sideLength + "x" + sideLength;
        int totalCells = sideLength * sideLength;
        return String.format("Input numbers between 1-%s on alternative turns to place your mark in the %s grid.", totalCells, gridShape);
    }

    static String getIntro() {
        return "TIC TAC TOE" + "\n" +
                "The classic game of noughts and crosses!\nTurn friends into enemies as 2 players take turns marking spaces in a grid.\nWin short-lived glory by succeeding in placing a complete line in any horizontal, vertical or diagonal direction." + "\n";
    }
}
